### ISSUE TITLE
[BUG Marlin 1.1.9] LCD cycles in and out of different menus, when heat bed is turned on using LCD/ pronterface

### ISSUE THREAD

**Issue by Fr33man85** [MarlinFirmware/Marlin#13940]
*sry.. thought the main thread was closed..
https://github.com/MarlinFirmware/Marlin/issues/12712

**Comment by mhsuhail** [comment]
The main thread was closed without any solution.

**Comment by Fr33man85** [comment]
Thanks for your reply.
i go ahead in this thread.

i try some other things, maybe i ask the support from Creality directly because it is also happening when i use the provided Firmware from they homepage (the Source files, not the Hex, because i flashed the bootloader)

but maybe thats the Problem?

Do you use a bootloader on your board? 
Which board your Prusa i3 clone use?

**Comment by InsanityAutomation** [comment]
This is a well documented issue in many Creality circles. Its caused by the wiring on the back of the lcd picking up noise. Most people have been adding tape across the pins in the control box, and tying down the lcd ribbon cable to prevent it. Could also possibly disable bed pid control to reduce noise and liklihood of this occuring.

**Comment by Fr33man85** [comment]
ah ok, 
but not so well documented in the general web :D

Do you have some links where i can read about it?

If this is a noise problem, maybe it can be solved to add some Pullup/down resistors to the corresponding Pins.

but anyway, i try to shield the ribbon cable instead

**Comment by Fr33man85** [comment]
o.O

Or i first try to chance the cable complete

![WhatsApp Image 2019-05-08 at 20 06 54](https://user-images.githubusercontent.com/50385199/57397420-f178be80-71cc-11e9-81bb-3985b2ff9a4f.jpeg)

**Comment by InsanityAutomation** [comment]
Looks like something found that cable a bit tasty to gnaw on! Its mostly on the Creality FB groups, but at least in your case hopefully its the obvious!

**Comment by mhsuhail** [comment]
Thanks for the lead, InsanityAutomation, will give it a try when I get some time. Manuel I will appreciate if you can update me as well, if this works out for you. Thanks

**Comment by Fr33man85** [comment]
So, a few frustrating hours later:

First, I renewed the cable. But without success.

![IMG_20190508_214415](https://user-images.githubusercontent.com/50385199/57405759-8b496700-71df-11e9-8fc8-761c348d84aa.jpg)

Afterwards I thought about the statement of InsanityAutomation. This noise ... So I went in search of appropriate documentation on the pinout of the LCD.

But I did not find much, so I took the whole thing apart and measured myself.

The button does nothing else as pull a signal from a dedicated pin to ground. So the easiest thing to do was to put in a pullup resistor.

![IMG_20190508_214049](https://user-images.githubusercontent.com/50385199/57405821-af0cad00-71df-11e9-8b44-c1468be01f2b.jpg)

In the first attempt, I took a 10k.
Without putting everything together again, it looked good at first. The behavior did not recur.
I was already happy and put everything back together.
After that - oh, miracles - it happened again. This time, however, not nearly so bad.

So dismantled and changed the 10k to 4,7k.
Also tested in disassembled condition; looked good.
But here too: after assembly, it happened again.. grrrr

![IMG_20190508_220424](https://user-images.githubusercontent.com/50385199/57405874-c9df2180-71df-11e9-8f41-045195a11187.jpg)

I think there might be an earth loop problem somewhere or something. Next, I will probably try to improve the grounding as a whole.


:Edit:
If the display is not mounted to the frame, i can measure with the osciloscope an rising edge of about 8-10V from the frame to the display"housing". 
If it is not mounted, no problem.

If i mount it (or only touch the frame without screwed it together) the menu hopping starts.
(but the readings on the DSO go to zero) 
![IMG_20190508_230741](https://user-images.githubusercontent.com/50385199/57408696-9784f280-71e6-11e9-9ca8-d8e27c082765.jpg)
![IMG_20190508_230726](https://user-images.githubusercontent.com/50385199/57408823-f6e30280-71e6-11e9-900f-ff800bc04d29.jpg)

**Comment by Fr33man85** [comment]
So,
I've added a few ground wires and I suggest everyone do the same or at least check it!
The entire frame (metal, conductive) is NOT grounded. The negative output of the power supply is also NOT connected to ground.

I'm not an expert, but I think that should not be so.

I also stripped a CAT5e Ethernet cable to collect the shield and put the recycled shield on the ribbon cable.

since 3h printing time the "auto-menu-click" has not happened again.

![IMG_20190510_182409_1024x1024_500KB](https://user-images.githubusercontent.com/50385199/57549632-60d4e680-7364-11e9-99ee-69c95666c7f6.jpg)
![IMG_20190510_182420_1024x1024_500KB](https://user-images.githubusercontent.com/50385199/57549633-616d7d00-7364-11e9-9aa9-ec32ae2cde68.jpg)

![IMG_20190510_182450_1024x1024_500KB](https://user-images.githubusercontent.com/50385199/57549635-616d7d00-7364-11e9-9a4e-db6ba05d9308.jpg)
![IMG_20190510_111126_1024x1024_500KB](https://user-images.githubusercontent.com/50385199/57549672-7f3ae200-7364-11e9-9ec2-f9b93c1e969d.jpg)
